libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/NavigationHelperTest.java

```diff
@@ -133,35 +123,344 @@ public void offRouteDetectionDisabled_isOffRouteReturnsFalse() throws Exception
     MapboxNavigation mapboxNavigation = new MapboxNavigation(mock(Context.class), ACCESS_TOKEN, options,
       mock(NavigationTelemetry.class), mock(LocationEngine.class));
     NewLocationModel model = NewLocationModel.create(mock(Location.class), mapboxNavigation);
-    assertFalse(isUserOffRoute(model, mock(RouteProgress.class), mock(OffRouteCallback.class)));
+
+    boolean userOffRoute = isUserOffRoute(model, mock(RouteProgress.class), mock(OffRouteCallback.class));
+
+    assertFalse(userOffRoute);
   }
 
   @Test
   public void stepDistanceRemaining_returnsZeroWhenPositionsEqualEachOther() throws Exception {
+    DirectionsRoute route = buildMultiLegRoute();
     Point snappedPoint = Point.fromLngLat(-77.062996, 38.798405);
     List<Point> coordinates = PolylineUtils.decode(
       route.legs().get(0).steps().get(1).geometry(), Constants.PRECISION_6);
+
     double distance = NavigationHelper.stepDistanceRemaining(snappedPoint, 0,
       1, route, coordinates);
+
     assertEquals(0.0, distance);
   }
 
   @Test
   public void nextManeuverPosition_correctlyReturnsNextManeuverPosition() throws Exception {
+    DirectionsRoute route = buildMultiLegRoute();
     List<Point> coordinates = PolylineUtils.decode(
       route.legs().get(0).steps().get(0).geometry(), Constants.PRECISION_6);
+
     Point nextManeuver = NavigationHelper.nextManeuverPosition(0,
       route.legs().get(0).steps(), coordinates);
+
     assertTrue(nextManeuver.equals(route.legs().get(0).steps().get(1).maneuver().location()));
   }
 
   @Test
   public void nextManeuverPosition_correctlyReturnsNextManeuverPositionInNextLeg() throws Exception {
+    DirectionsRoute route = buildMultiLegRoute();
     int stepIndex = route.legs().get(0).steps().size() - 1;
     List<Point> coordinates = PolylineUtils.decode(
       route.legs().get(0).steps().get(stepIndex).geometry(), Constants.PRECISION_6);
+
     Point nextManeuver = NavigationHelper.nextManeuverPosition(stepIndex,
       route.legs().get(0).steps(), coordinates);
+
     assertTrue(nextManeuver.equals(route.legs().get(1).steps().get(0).maneuver().location()));
   }
+
+  @Test
+  public void createIntersectionList_returnsCompleteIntersectionList() throws Exception {
```

Comment:
There are some blocks of duplicated code 👇 
What about extracting them into `private` "factory" methods and used them across the tests?

Reply:
If you're talking about:
```
    RouteProgress routeProgress = buildMultiLegRouteProgress();
    LegStep currentStep = routeProgress.currentLegProgress().currentStep();
    LegStep upcomingStep = routeProgress.currentLegProgress().upComingStep();
```
This can't be extracted, as we need both current and upcoming steps for the `Arrange`